subprojects/core/src/main/java/org/gradle/caching/configuration/internal/BuildCacheConfigurationInternal.java

```diff
@@ -16,10 +16,11 @@
 
 package org.gradle.caching.configuration.internal;
 
-import org.gradle.caching.BuildCacheServiceFactory;
+import org.gradle.caching.BuildCacheService;
 import org.gradle.caching.configuration.BuildCache;
 import org.gradle.caching.configuration.BuildCacheConfiguration;
 
 public interface BuildCacheConfigurationInternal extends BuildCacheConfiguration {
-    <T extends BuildCache> BuildCacheServiceFactory<T> getFactory(Class<? extends T> configuration);
+    // Finds a build cache implementation class for the given configuration object
+    <T extends BuildCache> Class<? extends BuildCacheService> getBuildCacheServiceType(T configuration);
```

Comment:
Do we need to bother with the generic type `T` at all? I think this method could accept a `BuildCache` directly.